python/mxnet/gluon/model_zoo/vision/mobilenet.py

```diff
@@ -0,0 +1,145 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable= arguments-differ
+"""MobileNet, implemented in Gluon."""
+__all__ = ['MobileNet', 'mobilenet1_0', 'mobilenet0_75', 'mobilenet0_5', 'mobilenet0_25',
+           'get_mobilenet']
+
+from ....context import cpu
+from ...block import HybridBlock
+from ... import nn
+
+# Helpers
+def _add_conv(out, channels=1, kernel=1, stride=1, pad=0, num_group=1):
+    out.add(nn.Conv2D(channels, kernel, stride, pad, groups=num_group, use_bias=False))
+    out.add(nn.BatchNorm(scale=False))
+    out.add(nn.Activation('relu'))
+
+def _add_conv_dw(out, channels, stride):
+    _add_conv(out, channels=channels)
+    _add_conv(out, channels=channels, kernel=3, stride=stride, pad=1, num_group=channels)
+
+
+# Net
+class MobileNet(HybridBlock):
+    r"""MobileNet model from the
+    `"MobileNets: Efficient Convolutional Neural Networks for Mobile Vision Applications"
+    <https://arxiv.org/abs/1704.04861>`_ paper.
+
+    Parameters
+    ----------
+    classes : int, default 1000
+        Number of classes for the output layer.
+    """
+    def __init__(self, multiplier=1.0, classes=1000, **kwargs):
+        super(MobileNet, self).__init__(**kwargs)
+        with self.name_scope():
+            self.features = nn.HybridSequential(prefix='')
+            with self.features.name_scope():
+                _add_conv(self.features, channels=int(32*multiplier), kernel=3, pad=1, stride=2)
+                _add_conv(self.features, channels=int(32*multiplier), kernel=3, pad=1,
```

Comment:
if we use depth_wise first in _add_conv_dw, then we can skip the extra _add_conv here and the one at line 70?